src/CodeConnect.Touch/TouchControl.xaml.cs

```diff
@@ -78,36 +78,42 @@ public TouchControl(TouchBranchCommand entryCommand, Window parentWindow)
                 var branchCommand = command as TouchBranchCommand;
                 if (vsCommand != null)
                 {
-                    path.TouchUp += (s, e) =>
-                    {
-                        e.Handled = true;
-                        VisualStudioModule.ExecuteCommand(vsCommand.VsCommandName, vsCommand.VsCommandParams);
-                        parentWindow.Hide();
-                    };
+                    Action<RoutedEventArgs> handle = (RoutedEventArgs args) =>
+                   {
+                       args.Handled = true;
+                       VisualStudioModule.ExecuteCommand(vsCommand.VsCommandName, vsCommand.VsCommandParams);
+                       parentWindow.Hide();
+                   };
+
+                    path.MouseLeftButtonUp += (s, e) => { handle(e); };
+                    path.TouchUp += (s, e) => { handle(e); };
                 }
                 else if (branchCommand != null)
                 {
-                    path.TouchUp += (s, e) =>
+                    Action<RoutedEventArgs, Point> handle = (RoutedEventArgs args, Point position) =>
                     {
-                        e.Handled = true;
+                        args.Handled = true;
                         parentWindow.Hide();
-                        Show(branchCommand, e);
+                        Show(branchCommand, windowPosition);
                     };
+
+                    path.MouseLeftButtonUp += (s, e) => { handle(e, e.GetPosition(null)); };
+                    path.TouchUp += (s, e) => { handle(e, e.GetTouchPoint(null).Position); };
                 }
 
                 index++;
             }
         }
 
+        static Point windowPosition = new Point(0, 0);
```

Comment:
Move up to where other fields are and add a comment, eg `Previous position of the window, used do draw control for deeper levels in the same spot.`